### PR TITLE
Docs: Split function categories into different pages.

### PR DESCRIPTION
**Pull request by etefera**
Addresses https://github.com/GoogleContainerTools/kpt/issues/1000

### CODE REVIEW

**Review by monopole — APPROVED** [review]
/lgtm

File: site/content/en/guides/consumer/function/catalog/_index.md [hunk 1]
```diff
@@ -7,10 +7,6 @@ description: >
     Catalog of Config Functions.
```
Comment (prachirp) [inline site/content/en/guides/consumer/function/catalog/_index.md:7]: Remove this page and combine remaining two sentences into https://googlecontainertools.github.io/kpt/guides/consumer/function/

And revisit references to the catalog page from elsewhere on the site to see if they should link to top-level page or one of the new pages (https://googlecontainertools.github.io/kpt/search/?q=catalog)

**Review by prachirp — CHANGES_REQUESTED** [review]
Nice job breaking this up into different pages. Have some suggestions.

File: site/content/en/guides/consumer/function/catalog/catalog/index_generator.js [hunk 2]
```diff
@@ -26,67 +26,48 @@ function tableByType(type) {
...
+weight: 7
+type: docs
+description: >
+    Catalog of ${name} Functions.
+---
+<!---
+DO NOT EDIT. Generated by: "cd ../catalog; npm run gen-docs"
+-->
+${descriptions[name.toLowerCase()] || ""}
+## ${name} functions
```
[the remark this answers is not in the document]
Comment (etefera) [reply site/content/en/guides/consumer/function/catalog/catalog/index_generator.js]: Fixed.

File: site/content/en/guides/consumer/function/catalog/catalog/index_generator.js [hunk 3]
```diff
@@ -26,67 +26,48 @@ function tableByType(type) {
...
+}
 
-${tableByType('transformer')}
+const name = process.argv[2]
 
-## Miscellaneous
+const README = `---
+title: "${name} Functions Catalog"
+linkTitle: "${name} Functions Catalog"
+weight: 7
```
[the remark this answers is not in the document]
Comment (etefera) [reply site/content/en/guides/consumer/function/catalog/catalog/index_generator.js]: Fixed.

File: site/content/en/guides/consumer/function/catalog/catalog/index_generator.js [hunk 4]
```diff
@@ -26,67 +26,48 @@ function tableByType(type) {
...
-Run functions either imperatively or declaratively by following the
-[Functions User Guide].
-
-Implement configuration functions using any toolchain such as the
-[Typescript SDK][ts sdk] or [Golang Libraries][go libs].
-
-## Sources
-
-See [definition of source functions][source].
+const descriptions = {
```
[the remark this answers is not in the document]
Comment (frankfarzan) [reply site/content/en/guides/consumer/function/catalog/catalog/index_generator.js:29]: I would avoid using 'source functions' and 'sink functions' at all in the catalog. We're going to make a conceptual change for those anyways.

**Review by frankfarzan — APPROVED** [review]

File: site/content/en/guides/consumer/function/catalog/catalog/index_generator.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (etefera) [reply site/content/en/guides/consumer/function/catalog/catalog/index_generator.js]: Noting that this means the individual categories will appear below the Exporting a Workflow option.

File: site/content/en/guides/consumer/function/catalog/_index.md (hunk 1 above)
Comment (etefera) [reply site/content/en/guides/consumer/function/catalog/_index.md]: Done.

File: site/content/en/guides/consumer/function/catalog/catalog/index_generator.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (etefera) [reply site/content/en/guides/consumer/function/catalog/catalog/index_generator.js]: Added some starter descriptions. Are there any suggestions for those points of confusion? Offered up something for validation.

**Review by prachirp — APPROVED** [review]
LGTM. Just added two more function [examples](https://github.com/GoogleContainerTools/kpt-functions-catalog/tree/master/examples) if you want to add the links into this change.

**Comment by monopole** [comment]
Fixes #1000

**Pull request merged**